### PR TITLE
Fix cognite_modules Warnings

### PR DESCRIPTION
**Pull request by doctrino**
## Description
We have the following warnings which are fixed by this PR except for the one taken care of in #219 
```
      WARNING: the space demo_project\cognite_modules\examples\cdf_apm_simple_data_model\data_models\apm_simple.space.yaml has an externalId apm_simple without the recommended 'sp_' based prefix.
  WARNING: Found missing data_sets:
    In file 'build\\transformations\\1.tr_asset_oid_workmate_apm_simple_load_asset2children.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 1.tr_asset_oid_workmate_apm_simple_load_asset2children.yaml. Did you forget to add it?
  WARNING: Found missing data_sets:
    In file 'build\\transformations\\2.tr_asset_oid_workmate_apm_simple_load_assets.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 2.tr_asset_oid_workmate_apm_simple_load_assets.yaml. Did you forget to add it?
  WARNING: Unresolved template variable {{source_files}} in build\transformations\tr_timeseries_oid_pi_apm_simple_load_timeseries2assets.sql
  WARNING: Found missing data_sets:
    In file 'build\\transformations\\3.tr_timeseries_oid_pi_apm_simple_load_timeseries2assets.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 3.tr_timeseries_oid_pi_apm_simple_load_timeseries2assets.yaml. Did you forget to add it?
  WARNING: Found missing data_sets: 
    In file 'build\\transformations\\4.tr_workorder_oid_workmate_apm_simple_load_workitems.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 4.tr_workorder_oid_workmate_apm_simple_load_workitems.yaml. Did you forget to add it?
  WARNING: Found missing data_sets:
    In file 'build\\transformations\\5.tr_workorder_oid_workmate_apm_simple_load_workitems2assets.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 5.tr_workorder_oid_workmate_apm_simple_load_workitems2assets.yaml. Did you forget to add it?
  WARNING: Found missing data_sets: 
    In file 'build\\transformations\\6.tr_workorder_oid_workmate_apm_simple_load_workitems2workorders.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 6.tr_workorder_oid_workmate_apm_simple_load_workitems2workorders.yaml. Did you forget to add it?
  WARNING: Found missing data_sets:
    In file 'build\\transformations\\7.tr_workorder_oid_workmate_apm_simple_load_workorders.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 7.tr_workorder_oid_workmate_apm_simple_load_workorders.yaml. Did you forget to add it?
  WARNING: Found missing data_sets:
    In file 'build\\transformations\\8.tr_workorder_oid_workmate_apm_simple_load_workorders2assets.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 8.tr_workorder_oid_workmate_apm_simple_load_workorders2assets.yaml. Did you forget to add it?
  WARNING: Found missing data_sets: 
    In file 'build\\transformations\\9.tr_asset_oid_workmate_asset_hierarchy.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 9.tr_asset_oid_workmate_asset_hierarchy.yaml. Did you forget to add it?
  WARNING: Found missing data_sets: 
    In file 'build\\transformations\\10.tr_asset_oid_workmate_asset_hierarchy.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 10.tr_asset_oid_workmate_asset_hierarchy.yaml. Did you forget to add it?
      WARNING: the transformation demo_project\cognite_modules\examples\example_pump_asset_hierarchy\transformations\pump_asset_hierarchy-load-collections_pump.yaml has an externalId pump_asset_hierarchy-load-collections_pump without the  
recommended 'tr_' based prefix.
  WARNING: Found missing data_sets:
    In file 'build\\transformations\\11.pump_asset_hierarchy-load-collections_pump.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 11.pump_asset_hierarchy-load-collections_pump.yaml. Did you forget to add it?
      WARNING: the space demo_project\cognite_modules\experimental\cdf_asset_source_model\data_models\1.InstanceSpace.space.yaml has an externalId cdfTemplateInstances without the recommended '_' based namespacing.
      WARNING: the space demo_project\cognite_modules\experimental\cdf_asset_source_model\data_models\2.ModelSpace.space.yaml has an externalId ExtendedSourceDataModels without the recommended '_' based namespacing.
      WARNING: the transformation demo_project\cognite_modules\experimental\cdf_asset_source_model\transformations\sync-asset_hierarchy_cdf_asset_source_model.yaml has an externalId sync-asset_hierarchy_cdf_asset_source_model without the 
recommended 'tr_' based prefix.
  WARNING: Found missing data_sets: 
    In file 'build\\transformations\\12.sync-asset_hierarchy_cdf_asset_source_model.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 12.sync-asset_hierarchy_cdf_asset_source_model.yaml. Did you forget to add it?
      WARNING: the space demo_project\cognite_modules\experimental\example_pump_data_model\data_models\1.spaces.space.yaml has an externalId pumpInstanceSpace without the recommended '_' based namespacing.
      WARNING: the space demo_project\cognite_modules\experimental\example_pump_data_model\data_models\1.spaces.space.yaml has an externalId pumpModelSpace without the recommended '_' based namespacing.
      WARNING: the transformation demo_project\cognite_modules\experimental\example_pump_data_model\transformations\pump_model-populate-lift_station_pumps_edges.yaml has an externalId pump_model-populate-lift_station_pumps_edges without 
the recommended 'tr_' based prefix.
  WARNING: Found missing data_sets:
    In file 'build\\transformations\\13.pump_model-populate-lift_station_pumps_edges.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 13.pump_model-populate-lift_station_pumps_edges.yaml. Did you forget to add it?
      WARNING: the transformation demo_project\cognite_modules\experimental\example_pump_data_model\transformations\pump_model-populate-pump_container.yaml has an externalId pump_model-populate-pump_container without the recommended 'tr_' 
based prefix.
  WARNING: Found missing data_sets:
    In file 'build\\transformations\\14.pump_model-populate-pump_container.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 14.pump_model-populate-pump_container.yaml. Did you forget to add it?
  WARNING: Found missing data_sets: 
    In file 'build\\transformations\\15.tr_asset_oid_workmate_infield_sync_assets_from_hierarchy_to_apm.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 15.tr_asset_oid_workmate_infield_sync_assets_from_hierarchy_to_apm.yaml. Did you forget to add it?
  WARNING: Found missing data_sets:
    In file 'build\\transformations\\16.tr_workorder_oid_workmate_infield_sync_workorders_to_apm_activities.yaml'
        DataSetMissingWarning: It is recommended that you set dataSetExternalId for Transformation. This is missing in 16.tr_workorder_oid_workmate_infield_sync_workorders_to_apm_activities.yaml. Did you forget to add it?
```

## Checklist:
- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped. [_version.py](https://github.com/cognitedata/cdf-project-templates/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/cdf-project-templates/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/cognite_modules/examples/cdf_apm_simple_data_model/transformations/tr_timeseries_oid_pi_apm_simple_load_timeseries2assets.sql [hunk 1]
```diff
@@ -2,4 +2,4 @@ select
   cast(`asset` as STRING) as externalId,
   array(timeseries) as metrics
 from
-  `files_{{default_location}}_{{source_files}}`.`timeseries2assets`;
+  `files_{{default_location}}_{{source_timeseries}}`.`timeseries2assets`;
```
Comment (gregertw) [inline cognite_toolkit/cognite_modules/examples/cdf_apm_simple_data_model/transformations/tr_timeseries_oid_pi_apm_simple_load_timeseries2assets.sql:5]: Good catch!

File: cognite_toolkit/cognite_modules/experimental/cdf_asset_source_model/default.config.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Only valid for this module, loads template variables from environment
-model_space: ExtendedSourceDataModels
-instance_space: cdfTemplateInstances
+model_space: sp_ExtendedSourceDataModels
```
Comment (gregertw) [inline cognite_toolkit/cognite_modules/experimental/cdf_asset_source_model/default.config.yaml:2]: I have commented on this before with mixed case approaches (which I don't think is a good idea), and the naming convention says that _ should be used as separator, so sp_extended_source_data_models.

File: cognite_toolkit/cognite_modules/experimental/example_pump_data_model/default.config.yaml [hunk 3]
```diff
@@ -1,9 +1,10 @@
 # Only valid for this module, loads template variables from environment
-model_space: pumpModelSpace
-instance_space: pumpInstanceSpace
-source_model_space: ExtendedSourceDataModels
+model_space: sp_pumpModelSpace
```
Comment (gregertw) [inline cognite_toolkit/cognite_modules/experimental/example_pump_data_model/default.config.yaml:2]: Same here with casing.

File: cognite_toolkit/cognite_modules/infield/cdf_infield_location/transformations/tr_asset_oid_workmate_infield_sync_assets_from_hierarchy_to_apm.yaml [hunk 4]
```diff
@@ -10,6 +10,7 @@ destination:
 ignoreNullFields: true
 shared: true
 action: upsert
+dataSetExternalId: ds_infield_{{default_location}}_app_data
```
Comment (gregertw) [inline cognite_toolkit/cognite_modules/infield/cdf_infield_location/transformations/tr_asset_oid_workmate_infield_sync_assets_from_hierarchy_to_apm.yaml:13]: So, this is not correct now. This is not the app data data set, but rather the default location asset data set, ds_asset_{{default_location}}_source_data if I recall correctly. But this is something @erikangelsen-cognite probably should quality assure.

File: cognite_toolkit/cognite_modules/infield/cdf_infield_location/transformations/tr_workorder_oid_workmate_infield_sync_workorders_to_apm_activities.yaml [hunk 5]
```diff
@@ -11,6 +11,7 @@ destination:
 ignoreNullFields: true
 shared: true
 action: upsert
+dataSetExternalId: ds_infield_{{default_location}}_app_data
```
Comment (gregertw) [inline cognite_toolkit/cognite_modules/infield/cdf_infield_location/transformations/tr_workorder_oid_workmate_infield_sync_workorders_to_apm_activities.yaml:14]: Here I believe we shouldn't have a data set. If I remember correctly, this is a RAW to data model transformation that does not touch a data set.

**Review by gregertw — COMMENTED** [review]
This is a good clean up! We need the infield team to review, ref my comments.

File: cognite_toolkit/cognite_modules/infield/cdf_infield_location/transformations/tr_asset_oid_workmate_infield_sync_assets_from_hierarchy_to_apm.yaml (hunk 4 above)
Comment (erikangelsen-cognite) [reply cognite_toolkit/cognite_modules/infield/cdf_infield_location/transformations/tr_asset_oid_workmate_infield_sync_assets_from_hierarchy_to_apm.yaml:13]: @gregertw yes you are correct, assets is ingested from their source system, so it should end up in the "source data" spaces and data sets.

File: CHANGELOG.templates.md [hunk 6]
```diff
@@ -41,6 +43,8 @@ Changes are grouped as follows:
 ### Fixed
 
 - Removed transformation identity provider variables from modules and reused the global cicd_ prefixed ones.
+- In the `cdf_infield_location` set dataset for transformations.
```
Comment (doctrino) [inline CHANGELOG.templates.md:46]:
```suggestion
```

**Review by gregertw — APPROVED** [review]

**Pull request merged**